### DISCUSSION TITLE
Progress Updates and Checkins

### DISCUSSION THREAD

**Discussion by tkbdsi** [thkruz/the-enhanced-scheduling-tool#17] [General]
February 20

I made a lot of adjustments to code that broke after the template refactors on styling.

I started working on the main Calendar component which requires a revisit to the useFetch. At the moment, while the code works, there is a pipeline issue in the CI/CD workflow. It doesn't like that I have arguments passed to useFetch that also don't end up in the dependency array. I will find a work around for this in a way that doesn't fry everyone's computers with infinite loops.

-----
February 19

Plan: hanging out with the family today. Will likely revisit the main calendar page later this evening (circa 8 PMish).

A lot of my night actually gotten eaten up with the State travel website. Kept crashing on me (by now my passport number is emblazoned onto the back of my retina-- I had to retype it so many times).

Managed to get on and help with some approvals for merging Ted's touches. Really like the restyles and agree by using a styling framework, it will help speed up so we can just focus on the coding.

-----
February 18

-update 2: Well... my pull request works all the way til it gets to SonarCube. Then it needs an environment variable. That's something I cant fix, since its not related to the React code itself. I'll just work on some Calendar page stuff, and do what I can, but Ted will have to be the one to fix the last crashing item in the github Actions routine.

-Update 1: put together a sketched app layout plus finished the Styled Components. I've submitted a Pull request for review, but I don't see any breaking changes since I made my stuff fairly isolated.

-- Started working on a Loading component today. I created my own branch for it. When I went to push an update, I wasn't able too. I think it has something to do with however I am configured with access to the repo. I might have to instead fork the repo and start over.

**Comment by thkruz** [comment]
## February 19
I found that I needed to run ```npm i``` in order to get everything working again. Suspect you figured it out based on the PRs though.

I got the astrouxds stuff working today. PITA but I think having their CSS framework and some of their components will make this more about writing code and less about doing design.

Would recommend that we scrap the app drawer that slides out and just leave it static on the left. MUI recommends hiding them on mobile and leaving them visible on desktops (which is our only target audience).

I have been coding 12 hours a day all week so my wrists need a break. I will start tackling issues tomorrow afternoon or Monday. I think this is moving along at a great pace. I did some work building cypress tests and integrating them into Gitlab/Github pipelines this week so I hopefully can add those to our pipeline this week so that we have unit and e2e tests.

**Reply by tkbdsi** [reply]
Excellent. I was going to mention maybe installing cypress on the next big merge, but looks like you are already on it.

Yeah man, that's a lot of full days of coding! Definitely a break earned. I also agree on the just leaving the lefthand side static (and we can always just hide out the righthand side completely... I just sort of sketched it in based on what I saw from our inspiration source).

Need some feedback on how we want the schedule.json file to look. Since the Calendar component will fetch by post to the calendar route on the server, looks like we have to create that schedule.json file. I liked your initial idea of just pulling that schedule from the server and "showing it" (since this whole project is just about showing stuff). We can always put it as a stretch to modify it if we are keeping up a strong pace.

If no one has strong feelings about it, I can just draft something up. But someone might already have a vision of what they wanted.... ;-)

**Reply by thkruz** [reply]
> Need some feedback on how we want the schedule.json file to look

On it

**Reply by tkbdsi** [reply]
> > Need some feedback on how we want the schedule.json file to look
> 
> 
> 
> On it

Thanks! (Mobile right now, but I'll take a look later)

**Reply by tkbdsi** [reply]
I've spent the first leg of this trip trying to get a useFetch.test to work. I can actually pass a test for it, but I still get all sorts of "act" warnings, which tells me that even when I use "act" from the test-utils, I am not using it in some other place in the right way most likely.

These warnings will likely crash the pipeline, so I am going to leave the useFetch.test out of my commits for now (adding it to the gitignore file so I don't have to delete it completely).

**Comment by tkbdsi** [comment]
Everyone might need to make a pull soon before they do any work this weekend.

Logic Flow
Ted set up the Context, closing out that feature developer issue. Using that, I went through and updated a lot of the logic by refactoring states to use this context after wrapping the app in a provider. This was super important when it came to the Admin page, since adding and removing users cascades changes through all the other pages.

Admin Page
The main features were test driven developed. The roster can be manipulated to add and remove users from context. I also added logic to download the roster json to file.

On Styling
I fixed a lot of styling throughout, following the chill UI style that Ted picked out. I also updated a lot of the issues where it looked like content was spilling it guts out when monitor resolution wasn't on an epic monitor (looking at you small laptops).

Next Steps
Only two really pending issues. I think we really need the MemberDay and ShiftDays to be implemented soonest now. That way we can integrate them into the established components. Also need to get cypress flowing, to be that end to end meat between all the unit tests.

I'm totally stoked with where we are. There is just so much cool logic in this frontend, I'm even happier with Ted's concept now that I'm seeing the end of the light in the tunnel haha.

My Status
My sister gets married in a few hours now. I'll be out for a bit, but can check back in tomorrow to see what's what :-)
